docs/pages/machine-id/guides/github-actions.mdx

```diff
@@ -0,0 +1,195 @@
+---
+title: Using Machine ID With GitHub Actions
+description: A tutorial for using Machine ID with GitHub Actions
+---
+
+GitHub Actions are a popular CI/CD platform that works as a larger part of the GitHub
+ecosystem. Teleport, with the help of Machine ID, allows for GitHub Actions to
+securely interact with Teleport Protected resources without the need for long lived
+credentials. By ensuring that rotated Machine ID credentials are additionally limited by RBAC,
+GitHub Actions with Teleport can be applied as both a viable and secure solution whenever a
+CI/CD platform is needed.
+
+<Admonition type="note">
+Currently GitHub Actions with Machine ID are unsupported on GitHub Enterprise
+or self-hosted runners.
+</Admonition>
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- A node that is a part of the Teleport cluster with the SSH service installed.
+- A machine with `tsh` access to the cluster.
+- A GitHub repository with GitHub Actions enabled. This guide uses the example `gravitational/example`
+repo, however this value should be replaced with your own unique repo.
+
+## Step 1/3. Create a join Token for GitHub Actions
+
+GitHub Actions will need it's own join token in order to be able to join the cluster.
+Using a text editor of your choice, create a configuration YAML file `tokenconfig.yaml` for the
+join token that resembles the following:
+
+```yaml
+kind: token
+version: v2
+metadata:
+  name: github-demo
+  expires: "3000-01-01T00:00:00Z"
+spec:
+  roles: [Bot]
+  join_method: github
+  bot_name: github-demo
+  github:
+    allow:
+      - repository: gravitational/example
+```
+
+Let's go over the token configuration YAML in more detail:
+
+- `metadata.name` and `spec.bot_name` in this case represent the name of the
```

Comment:
`metadata.name` represents the name of the token that will be provided in `--token` later. It isn't necessarily the same as the bot name, although, in my own clusters I tend to have them matching unless I'm creating a bot that has multiple associated tokens.

Reply:
I'm thinking we should use a different token name in the config upon further consideration. I think in an example like this it's best for users to be able differentiate where possible, unless there's some specific reason we would want to keep the value the same.